sys/usb/usbus/Kconfig

```diff
@@ -9,6 +9,7 @@ menuconfig MODULE_USBUS
     bool "USB Unified Stack (USBUS)"
     depends on TEST_KCONFIG
     depends on HAS_PERIPH_USBDEV || MODULE_USBDEV_MOCK
+    default y if TEST_KCONFIG && HAS_HIGHLEVEL_STDIO && !MODULE_TINYUSB_DEVICE
```

Comment:
> I believe the natural way to model two mutually exclusive implementations in Kconfig would be a `choice`.

I agree with this in principle. The problem I see is that USBUS and its device classes are not configured in a separate submenu but are fully integrated into the `System` configuration menu.  On top level the user can select the USBUS
```
[ ] USB Unified Stack (USBUS)  ----
```
but if it is selected, the `System` configuration menu is extended to
```
[*] USB Unified Stack (USBUS)  --->
[ ] USB CDC ACM support (NEW)
[ ] USB HID support (NEW)  ----
```
and if `USB CDC ACM support` is selected the `System` configuration menu is further extended to
```
[*] USB Unified Stack (USBUS)  --->
[*] USB CDC ACM support
[ ] USB HID support  ----
[ ] Trigger a board reset via USB CDC ACM (NEW)
```
The tinyUSB configuration on the other hand starts in the `Packages` configuration menu, because it is a package. But unlike USBUS, tinyUSB configuration is done completely in separate submenus.

If we want to provide a more natural solution, there should be just an entry like
```
[ ] USB interface ----
```
in the `System` configuration menu, which should lead to a new submenu where the user first selects the implementation 
```
    USB implementation (USBUS) --->
[ ] ...
```
and all other configurations appear according to the selected implementation. The tinyUSB package would be an hidden option in that case and enabled automatically if tinyUSB implementation is selected.

Right?

Reply:
That is very true, we _should_ be able to do this with some choice overrides, in which case tinyusb would be a choice.  If you don't mind I would like to give an update to the usb modelling a try?